modules/aws/src/main/java/org/apache/ignite/spi/discovery/tcp/ipfinder/s3/TcpDiscoveryS3IpFinder.java

```diff
@@ -310,15 +312,27 @@ public void setClientConfiguration(ClientConfiguration cfg) {
     }
 
     /**
-     * Sets AWS credentials.
+     * Sets AWS credentials. Either use {@link #setAwsCredentials(AWSCredentialsProvider)} or this one.
      * <p>
      * For details refer to Amazon S3 API reference.
      *
      * @param cred AWS credentials.
      */
     @IgniteSpiConfiguration(optional = false)
     public void setAwsCredentials(AWSCredentials cred) {
-        this.cred = cred;
+        this.credProvider = new AWSStaticCredentialsProvider(cred);
```

Comment:
I would pass AWSCredentials directly to AmazonS3Client object letting Amazon lib to decide what kind of provider to use. It's clear that Amazon presently uses AWSStaticCredentialsProvider by default but the logic might change in the future and Ignite needs to be as robust as possible.